padlockcloud/server_test.go

```diff
@@ -503,6 +510,39 @@ func TestAuthentication(t *testing.T) {
 		res, _ = ctx.request("GET", ctx.host+"/activate/?t=asdf", "", ApiVersion)
 		testError(t, res, &BadRequest{"invalid activation token"})
 	})
+
+	t.Run("whitelist", func(t *testing.T) {
+		ctx.resetAll()
+
+		dir, err := ioutil.TempDir("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.RemoveAll(dir)
+
+		tmpFile := filepath.Join(dir, "tmpFile")
+		d1 := []byte(testEmail + "\n")
+		if err := ioutil.WriteFile(tmpFile, d1, 0644); err != nil {
+			t.Fatalf("Error writing to whitelist file: %s", err.Error())
+		}
+
+		//setup config with whitelist
+		ctx := newServerTestContextWithConfig(&ServerConfig{WhitelistPath: tmpFile})
```

Comment:
What you're doing here is declaring a new variable, not reassigning it. This is actually what we want in this case, since we want a 'specialised' server configuration for this test. May be a good idea to use a new variable name to avoid confusion though.